docs/world api.md

```diff
@@ -122,69 +122,75 @@ class RLWeb(WebWorld):
 
 ### MultiWorld Object
 
-The `MultiWorld` object references the whole multiworld (all items and locations
-for all players) and is accessible through `self.multiworld` inside a `World` object.
+The `MultiWorld` object references the whole multiworld (all items and locations for all players) and is accessible
+through `self.multiworld` from your `World` object.
 
 ### Player
 
-The player is just an integer in AP and is accessible through `self.player`
-inside a `World` object.
+The player is just an integer in AP and is accessible through `self.player` from your `World` object.
 
 ### Player Options
 
-Players provide customized settings for their World in the form of yamls.
-A `dataclass` of valid options definitions has to be provided in `self.options_dataclass`.
-(It must be a subclass of `PerGameCommonOptions`.)
-Option results are automatically added to the `World` object for easy access.
-Those are accessible through `self.options.<option_name>`, and you can get a dictionary of the option values via
-`self.options.as_dict(<option_names>)`, passing the desired options as strings. 
+Options are provided by the user as part of the generation process, intended to affect how their randomizer experience
+should play out. These can control aspects such as what locations should be shuffled, what items are in the itempool,
+etc. Players provide the customized options for their World in the form of yamls.
+
+A `dataclass`, which must be a subclass of `PerGameCommonOptions`, of valid options definitions has to be provided in
+`self.options_dataclass`. Option results are then automatically added to the `World` object for easy access, between
+`World` creation and `generate_early`. These are accessible through `self.options.<option_name>`, and you can get a
+dictionary with option values via `self.options.as_dict(<option_names>)`, passing the desired option names as strings.
+
+By convention, options are defined in `options.py` and will be used when parsing the players' yaml files.
+
+Each option has its own class, which inherits from a base option type, has a docstring to describe it, and a
+`display_name` property for display on the website and in spoiler logs.
+
+The actual name as used in the yaml is defined via the field names of a `dataclass` that is assigned to the world under
+`self.options_dataclass`. By convention, the strings that define your option names should be in `snake_case`.
+
+Common option types are `Toggle`, `DefaultOnToggle`, `Choice`, and `Range`.
+For more information, see the [options api doc](options%20api.md).
```

Comment:
The whole section is understandable, I'm just less sure about the dataclass paragraph but I was able to understand the rest